src/directed/bfs.rs

```diff
@@ -62,34 +62,41 @@ use std::iter::FusedIterator;
 ///                  |&p| p == GOAL);
 /// assert_eq!(result.expect("no path found").len(), 5);
 /// ```
-pub fn bfs<N, FN, IN, FS>(start: &N, successors: FN, success: FS) -> Option<Vec<N>>
+pub fn bfs<'a, N, S, FN, IN, FS>(start: S, successors: FN, success: FS) -> Option<Vec<N>>
 where
-    N: Eq + Hash + Clone,
+    N: Eq + Hash + Clone + 'a,
```

Comment:
I don't fully understand why this `+ 'a` is needed for `N`, but it seems to work 🤷